haros/extractor.py

```diff
@@ -824,12 +879,24 @@ def find_nodes(self, pkg):
         srcdir = self.package.path[len(self.workspace):]
         srcdir = os.path.join(self.workspace, srcdir.split(os.sep, 1)[0])
         bindir = os.path.join(self.workspace, "build")
-        parser = RosCMakeParser(srcdir, bindir, pkgs = self.packages,
-                                env = self.environment,
-                                vars = self._default_variables())
-        parser.parse(os.path.join(self.package.path, "CMakeLists.txt"))
-        self._update_nodelets(parser.libraries)
-        self._register_nodes(parser.executables)
+        cmake_path = os.path.join(self.package.path, "CMakeLists.txt")
+        if os.path.isfile(cmake_path):
+            parser = RosCMakeParser(srcdir, bindir, pkgs = self.packages,
+                                    env = self.environment,
+                                    vars = self._default_variables())
+            parser.parse(cmake_path)
+            self._update_nodelets(parser.libraries)
+            self._register_nodes(parser.executables)
+        else:
+            # It may be normal for pure Python project not to have a CMakeLists.txt
+            for file in pkg.source_files:
```

Comment:
Sorry about that.
In the new commit I changed it to only consider python files with a ``def main():`` function.
I think the current CMakeLists.txt parser also doesn't consider Python projects / entry points since it seems to only look for ``add_executable`` directives. 
It may be useful to also make it look for ``install`` and ``catkin_install_python`` directives.
It's not for this pull request, but please let me know what you think.